drivers/stepper/stepper_shell.c

```diff
@@ -0,0 +1,488 @@
+		entry->syntax = stepper_direction_map[idx].name;
+	} else {
+		entry->syntax = NULL;
+	}
+	entry->handler = NULL;
+	entry->help = "Stepper direction";
+	entry->subcmd = NULL;
+}
+
+SHELL_DYNAMIC_CMD_CREATE(dsub_stepper_direction, cmd_stepper_direction);
+
+static void cmd_stepper_microstep(size_t idx, struct shell_static_entry *entry)
+{
+	if (idx < ARRAY_SIZE(stepper_microstep_map)) {
+		entry->syntax = stepper_microstep_map[idx].name;
+	} else {
+		entry->syntax = NULL;
+	}
+	entry->handler = NULL;
+	entry->help = "Stepper microstep resolution";
+	entry->subcmd = NULL;
+}
+
+SHELL_DYNAMIC_CMD_CREATE(dsub_stepper_microstep, cmd_stepper_microstep);
+
+static void cmd_pos_stepper_motor_name(size_t idx, struct shell_static_entry *entry)
+{
+	const struct device *dev = shell_device_lookup(idx, NULL);
+
+	entry->syntax = (dev != NULL) ? dev->name : NULL;
+	entry->handler = NULL;
+	entry->help = "List Devices";
+	entry->subcmd = NULL;
+}
+
+SHELL_DYNAMIC_CMD_CREATE(dsub_pos_stepper_motor_name, cmd_pos_stepper_motor_name);
+
+static void cmd_pos_stepper_motor_name_dir(size_t idx, struct shell_static_entry *entry)
+{
+	const struct device *dev = shell_device_lookup(idx, NULL);
+
+	if (dev != NULL) {
+		entry->syntax = dev->name;
+	} else {
+		entry->syntax = NULL;
+	}
+	entry->handler = NULL;
+	entry->help = "List Devices";
+	entry->subcmd = &dsub_stepper_direction;
+}
+
+SHELL_DYNAMIC_CMD_CREATE(dsub_pos_stepper_motor_name_dir, cmd_pos_stepper_motor_name_dir);
+
+static void cmd_pos_stepper_motor_name_microstep(size_t idx, struct shell_static_entry *entry)
+{
+	const struct device *dev = shell_device_lookup(idx, NULL);
+
+	if (dev != NULL) {
+		entry->syntax = dev->name;
+	} else {
+		entry->syntax = NULL;
+	}
+	entry->handler = NULL;
+	entry->help = "List Devices";
+	entry->subcmd = &dsub_stepper_microstep;
+}
+
+SHELL_DYNAMIC_CMD_CREATE(dsub_pos_stepper_motor_name_microstep,
+			 cmd_pos_stepper_motor_name_microstep);
+
+static int parse_device_arg(const struct shell *sh, char **argv, const struct device **dev)
+{
+	*dev = device_get_binding(argv[arg_idx_dev]);
+	if (!*dev) {
+		shell_error(sh, "Stepper device %s not found", argv[arg_idx_dev]);
+		return -ENODEV;
+	}
+	return 0;
+}
+
+static int cmd_stepper_enable(const struct shell *sh, size_t argc, char **argv)
+{
+	const struct device *dev;
+	int err;
+	bool enable;
+
+	if (strcmp(argv[arg_idx_param], "on") == 0) {
+		enable = true;
+	} else if (strcmp(argv[arg_idx_param], "off") == 0) {
+		enable = false;
+	} else {
+		shell_error(sh, "Invalid enable value: %s", argv[arg_idx_param]);
+		return -EINVAL;
+	}
+
+	err = parse_device_arg(sh, argv, &dev);
+	if (err < 0) {
+		return err;
+	}
+
+	err = stepper_enable(dev, enable);
+	if (err) {
+		shell_error(sh, "Error: %d", err);
+	}
+
+	return err;
+}
+
+static int cmd_stepper_move(const struct shell *sh, size_t argc, char **argv)
+{
+	const struct device *dev;
+	int err;
+	int32_t micro_steps = strtol(argv[arg_idx_param], NULL, 0);
```

Comment:
have you considered using `shell_strtoul` instead of the library function directly? I think that one has better error handling and can be used to print a "invalid argument" error

Reply:
Did not know about these, neat 👍